src/internal_events/host_metrics.rs

```diff
@@ -0,0 +1,76 @@
+use metrics::counter;
+use vector_core::internal_event::InternalEvent;
+
+use super::prelude::{error_stage, error_type};
+
+#[derive(Debug)]
+pub struct HostMetricsScrapeError {
+    pub message: &'static str,
+}
+
+impl InternalEvent for HostMetricsScrapeError {
+    fn emit(self) {
+        error!(
+            message = self.message,
+            error_type = error_type::READER_FAILED,
+            stage = error_stage::RECEIVING,
```

Comment:
Unrelated to this change - but I wonder if there could be user confusion about the term `RECEIVING` for sources that scrape data rather than ingest data. Don't believe it's come up, so probably not but figured I'd throw it out there.

Reply:
Interesting. I had not thought of that. I would kind of think they are the same stage (could be wrong about that), but maybe the word "receiving" could be changed to be more encompassing?